Fail gracefully on malformed XML

Exit with an error message instead of panicking.

```shell
$ axmlfmt <(echo 'test')
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rsookram/axmlfmt/internal/parse.ReadXML(0x5068c0, 0xc0000b8000, 0xc0000b8000, 0x0, 0xc0000a4018, 0x0, 0x0)
	/path/to/axmlfmt/internal/parse/parse.go:63 +0xf02
main.main()
	/path/to/axmlfmt/cmd/axmlfmt/main.go:31 +0x142
```

--- a/internal/parse/parse.go
+++ b/internal/parse/parse.go
@@ -2,6 +2,7 @@ package parse
 
 import (
 	"encoding/xml"
+	"fmt"
 	"io"
 	"strings"
 )
@@ -60,6 +61,10 @@ func ReadXML(reader xml.TokenReader) ([]Element, error) {
 		case xml.CharData:
 			s := string(token)
 			if len(strings.TrimSpace(s)) != 0 {
+				if depth == 0 {
+					return nil, fmt.Errorf("unexpected top-level char data `%s`", s)
+				}
+
 				parent := stack[len(stack)-1]
 				parent.ContainsCharData = true
 

--- a/internal/parse/parse_test.go
+++ b/internal/parse/parse_test.go
@@ -23,6 +23,20 @@ func TestEmptyXML(t *testing.T) {
 	}
 }
 
+func TestTopLevelCharData(t *testing.T) {
+	doc := `test`
+
+	ee, err := read(doc)
+	if ee != nil {
+		t.Errorf("expected error, got %s", str(ee))
+	}
+
+	expected := "unexpected top-level char data `test`"
+	if err.Error() != expected {
+		t.Errorf("got %s, want %s", err.Error(), expected)
+	}
+}
+
 func TestElementContainingCommentNoChildren(t *testing.T) {
 	doc := `
 <shape

